beacon_node/network/src/sync/network_context/custody_by_root.rs

```diff
@@ -0,0 +1,417 @@
+use beacon_chain::BeaconChainTypes;
+use fnv::FnvHashMap;
+use lighthouse_network::service::api_types::{DataColumnsByRootRequestId, SingleLookupReqId};
+use lighthouse_network::PeerId;
+use lru_cache::LRUTimeCache;
+use rand::Rng;
+use slog::{debug, warn};
+use std::time::Duration;
+use std::{collections::HashMap, marker::PhantomData, sync::Arc};
+use types::EthSpec;
+use types::{data_column_sidecar::ColumnIndex, DataColumnSidecar, Hash256};
+
+use super::{
+    DataColumnsByRootSingleBlockRequest, LookupRequestResult, RpcResponseResult, SyncNetworkContext,
+};
+
+const FAILED_PEERS_CACHE_EXPIRY_SECONDS: u64 = 5;
+
+type DataColumnSidecarVec<E> = Vec<Arc<DataColumnSidecar<E>>>;
```

Comment:
For consistency:

```suggestion
type DataColumnSidecarList<E> = Vec<Arc<DataColumnSidecar<E>>>;
```

Also updated on the `das` branch `custody.rs`